src/api/forms/form-metadata-repository.js

```diff
@@ -0,0 +1,51 @@
+import * as fs from 'fs/promises'
+import { join } from 'path'
+import { config } from '~/src/config'
+import * as Types from '../types.js'
+
+// TODO look at prisma to generate this
+
+const formDirectory = config.get('formDirectory')
+
+/**
+ * Returns the filename for a form metadata entry, given a form ID.
+ * @param {string} formId ID of the form
+ * @returns {string} file path
+ */
+const getFormMetadataFilename = (formId) => {
+  return join(formDirectory, `${formId}-metadata.json`)
+}
+
+/**
+ * Retrieves a file from the form store
+ * @returns {Promise<Types.FormConfiguration[]>} form configuration
+ */
+const listForms = async () => {
+  return fs.readdir(formDirectory).then(async (files) => {
+    const response = []
+
+    for (const fileName of files) {
+      if (fileName.includes('-metadata.json')) {
+        const formId = fileName.replace('-metadata.json', '')
+        const fileJsonContent = await getFormMetadata(formId) // Added await here
+
+        response.push(fileJsonContent)
+      }
+    }
+
+    return response
+  })
```

Comment:
Not a blocker, but shall we add this metadata concurrently?

i.e. Instead of awaiting each file's metadata one by one

```suggestion
  const files = await readdir(formDirectory)

  const formIds = files
    .filter((fileName) => fileName.includes('-metadata.json'))
    .map((fileName) => fileName.replace('-metadata.json', ''))

  return Promise.all(formIds.map(getFormMetadata))
```